Add language variants knob to Caption

alex-magana opened the pull request:
Relates to #331, #370

Depends on #378  

**Overall change:** Adds the support for examining Psammead components' behaviour
in the different language services using the storybook select knob

**Code changes:**
- Adds stories to the `psammead-caption` component that use the select knob.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

DenisHernandez committed:
Add knobs addon to change the displayed text

DenisHernandez committed:
Bump version

alex-magana committed:
Add preset storybook knob text variants

alex-magana committed:
Refactor knob usage

alex-magana commented:
The `continuous-integration/jenkins/branch ` fails with
`npm ERR! Missing: @bbc/psammead-storybook-helpers@^0.1.0`
This should be resolved once #378 is merged and published.

alex-magana committed:
Bump package version

alex-magana committed:
Bump psammead version

dr3 commented on CHANGELOG.md:
```diff
@@ -2,6 +2,7 @@
 
 | Version | Description |
 |---------|-------------|
+| 1.2.0   | [PR#377](https://github.com/BBC/psammead/pull/377) Add select and text knobs in psammead-caption stories |
```
We dont need to bump/changelog for a change that doesnt reach outside of the `packages` dir :)

dr3 reviewed (dismissed):
LGTM after removing the changelog and installing the published package. Great work, it looks awesome

Commit:
Merge branch 'latest' into add-language-variants-knob

alex-magana committed:
Import inputProvider as a default export

alex-magana committed:
Import inputProvider from the base level

alex-magana committed:
Revert Psammead version bump

alex-magana committed:
Merge branch 'latest' into add-language-variants-knob

dr3 reviewed (commented):
LGTM, will need to wait for the other PR to be merged though

pjlee11 commented:
Blocked on #378

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into add-language-variants-knob

andrew-nowak committed:
Adjust dependency version

andrew-nowak committed:
Prevent over-escaping text knobs

By default, storybook will escape eg. apostrophes, causing input such as
`you'll` to be displayed as `you&#39;ll`. The only fix I could find was to
configure the knobs addon not to do this on every. Single. Story.

See also:
https://github.com/storybooks/storybook/issues/4445#issuecomment-459764918

andrew-nowak committed:
Empty commit to trigger snyk

https://support.snyk.io/hc/en-us/articles/360000920678-Failed-to-detect-issues

amywalkerdev reviewed (dismissed):
👍

Commit:
Merge branch 'latest' into add-language-variants-knob

dr3 commented on packages/components/psammead-caption/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@bbc/psammead-caption",
-  "version": "0.3.3",
+  "version": "0.4.0",
```
On second thoughts, this should be `0.3.4` as it doesn't affect the primary code

andrew-nowak replied:
Done

andrew-nowak committed:
Reduce version bump

andrew-nowak committed:
Merge remote-tracking branch 'origin/latest' into add-language-variants-knob

jamesbrumpton commented:
LGTM. Happy for this to be merged.
